### ISSUE TITLE
Rpi 1 model B - Not USB RC522 - GPIO buttons not working

### ISSUE THREAD

**Issue by dvalob** [MiczFlor/RPi-Jukebox-RFID#501]
Hi!

I am having troubles to use the GPIO buttons. I am using RPI 1 model B and a GPIO RFID reader and the non Spotify version. The reader is working fine and the gpio service is enabled and I see it with htop every 2/3 seconds. I also tried as said in #383 to put the pull_up to false. I also tried to comment out the non used commands and paid attention to GPIO/pin number mistake (#487). The GPIO numbers given in the script are not avail in my PI model, so I needed to change them.

I put the arcade buttons that you recommend as follows:
- Prev: pins 12 (GPIO18) - 14 (GND)
- Next: pins 9 (GND) - 11 (GPIO17)
- Pause: pins 25 (GND) -16 (GPIO23)

gpio script shows:
![image](https://user-images.githubusercontent.com/47637512/53427720-bfd3c200-39e9-11e9-98d3-3f9bfd7745cd.png)

GPIO and RIFD are enabled on web interface.

I checked with the commands gpio read XX but the values read never change.

gpio readall showns the following, the pins I used are correctly seen as IN (BCM column indicate the GPIO number, I tried also checking the wPI number. In fact I tried all the numbers :( ) I tried also putting v/3.3v on pin 11 (GPIO17) but gpio read 11 shows always 1. Are my GPIO working? a

![image](https://user-images.githubusercontent.com/47637512/53427583-800cda80-39e9-11e9-90ea-8d201025219c.png)

Any clue??

I guess that GPIO are working because RIFD reader is on it. But I have no more clue for the moment.
@hailogugo: help! :)

**Comment by dvalob** [comment]
I used a small script to see what is being read o this GPIO:

![image](https://user-images.githubusercontent.com/47637512/53430276-7c2f8700-39ee-11e9-8ab6-d77712adb41e.png)

And it happens that the Button press is seen, then when I release it, I see a toggling between pressed/not pressed. 

![image](https://user-images.githubusercontent.com/47637512/53430346-9bc6af80-39ee-11e9-9259-3e8d353c77f9.png)
I tried the same using a wire to simulate a contact/not contact and I have the same result. So it does not seem to be linked to the HW. Maybe the pull up resistor would solve this toggling value?

I suppose that the Rpi does not like to much to see this value toggling in order to go to next/previous track, etc.

Can anyone having these buttons working can run the given script and tell me if you have the same output?? @bofelix? @georgbachmann?

**Comment by Ardios** [comment]
@dvalob 

how many Pins have your Acarde Button? 2 or 3?

**Comment by dvalob** [comment]
I have 3 buttons, each one connected to 2 pins (one GPIO and one GND). I am not using a "common ground", but 3 different GND pins

**Comment by dvalob** [comment]
Hi! I finally had it working after several resets. The only new thing that I did is to comment all the unsused lines in the gpio-buttons.py (I had forgot lines  between 62 and 75). I do not use external pull-up resistors.

I also added a rotary switch for volume and I used its pushbutton (SW pin) to properly shutdown the PI.

At the end, my pin allocation is :

![image](https://user-images.githubusercontent.com/47637512/53723226-88409c00-3e67-11e9-81e3-d267486a24f4.png)

Daniel
